### PR TITLE
Adding group api

### PR DESCRIPTION
**Pull request by agreco**
Adding group api

### CODE REVIEW

**Comment by paulwib** [comment]
@agreco can you add some unit tests for this? Should be pretty much a copy of the documents API tests.

**Comment by agreco** [comment]
@paulwib I've committed the tests as requested.. pull request doesn't seem to have updated with this change, shall I create a new pull request?

File: test/spec/api/groups.spec.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+            it('should NOT have a body', function() {
+                expect(ajaxRequest.data).toBeUndefined();
+            });
+        });
+
+        describe('pagination', function() {
+
+            var sendMendeleyCountHeader = true,
+            folderCount = 56,
```
Comment (paulwib) [inline test/spec/api/groups.spec.js]: Can you rename `folderCount` to `groupCount` in this test?

**Comment by agreco** [comment]
@paulwib the groupCount change has been applied to the group spec

**Comment by paulwib** [comment]
cool thx

**Pull request merged**